### PR TITLE
#499: "flat" set as default layout

### PR DESCRIPTION
**Pull request by IlyaMoskva**
#499 Fixed and Reworked.

### CODE REVIEW

File: src/test/java/com/artipie/YamlSettingsTest.java [hunk 1]
```diff
@@ -46,12 +46,27 @@
...
 
+    @Test
+    public void shouldSetFlatAsDefaultLayout() throws Exception {
+        final YamlSettings settings = new YamlSettings(
+            String.join(
+                "",
+                "meta:\n",
+                "  storage:\n",
+                "    type: s3\n",
+                "    path: /var/artipie/configs\n"
```
Comment (olegmoz) [inline src/test/java/com/artipie/YamlSettingsTest.java:61]: @IlyaMoskva do we really need to provide all that content to `YamlSettings`? I'd suggest to remove not required lines from the test to make it more focused and clear

File: src/test/java/com/artipie/YamlSettingsTest.java [hunk 2]
```diff
@@ -46,12 +46,27 @@
...
+            )
+        );
+        MatcherAssert.assertThat(
+            settings.layout(),
+            Matchers.is("flat")
```
Comment (olegmoz) [inline src/test/java/com/artipie/YamlSettingsTest.java:66]: @IlyaMoskva using `new IsEqual<>()` here will be more in line with code style of the project, see https://github.com/artipie/artipie/blob/master/CONTRIBUTING.md#code-style

**Review by olegmoz — CHANGES_REQUESTED** [review]
@IlyaMoskva thanks, please see two comments

File: src/test/java/com/artipie/YamlSettingsTest.java (hunk 2 above)
Comment (IlyaMoskva) [reply src/test/java/com/artipie/YamlSettingsTest.java]: Changed.

File: src/test/java/com/artipie/YamlSettingsTest.java (hunk 1 above)
Comment (IlyaMoskva) [reply src/test/java/com/artipie/YamlSettingsTest.java]: Not all, you're right. Fixed.

File: src/test/java/com/artipie/YamlSettingsTest.java [hunk 3]
```diff
@@ -56,14 +57,12 @@ public void shouldSetFlatAsDefaultLayout() throws Exception {
             String.join(
                 "",
                 "meta:\n",
-                "  storage:\n",
-                "    type: s3\n",
-                "    path: /var/artipie/configs\n"
+                "  storage:\n"
```
Comment (olegmoz) [inline src/test/java/com/artipie/YamlSettingsTest.java:60]: @IlyaMoskva and is `storage:` part really needed? We do not access `storage` in this test and I believe the string could be simplified to just `"meta:\n"

**Review by olegmoz — CHANGES_REQUESTED** [review]
@IlyaMoskva thanks, please see one comment

File: src/test/java/com/artipie/YamlSettingsTest.java (hunk 3 above)
Comment (IlyaMoskva) [reply src/test/java/com/artipie/YamlSettingsTest.java:60]: It breaks in constructor or in other place before. Just 'meta:' doesn't work for me.

Comment (olegmoz) [reply src/test/java/com/artipie/YamlSettingsTest.java:60]: @IlyaMoskva that's weird, there is just assignment in constructor. Could you please elaborate the problem?

Comment (IlyaMoskva) [reply src/test/java/com/artipie/YamlSettingsTest.java:60]: @olegmoz Exactly storage is not necessary here, but empty meta throws Exception in .yamlMapping(YamlSettings.KEY_META) call. It must contain at least something inside. I believe, it's out of scope for this ticket.

Comment (olegmoz) [reply src/test/java/com/artipie/YamlSettingsTest.java:60]: @IlyaMoskva got it, thanks

**Review by olegmoz — APPROVED** [review]

**Comment by 0crat** [comment]
Job `gh:artipie/artipie#577` is not assigned, can't get performer

**Pull request merged**